src/session/sidebar/selection.rs

```diff
@@ -232,6 +232,10 @@ impl Selection {
         imp.item.replace(item);
 
         let old_position = imp.item_position.get();
+        if old_position == gtk::INVALID_LIST_POSITION && position == gtk::INVALID_LIST_POSITION {
+            return;
```

Comment:
Oops, we can't return early here because otherwise we don't notify `selected-item` below.